internal/cmd/options.go

```diff
@@ -0,0 +1,261 @@
+/*
+Copyright 2022 The OpenVEX Authors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openvex/go-vex/pkg/vex"
+	"github.com/spf13/cobra"
+)
+
+type vexDocOptions struct {
+	DocumentID string
+	Author     string
+	AuthorRole string
+}
+
+const (
+	productLongFlag = "product"
+	vulnLongFlag    = "vuln"
+)
+
+// Validate checks that the document options are valid
+func (do *vexDocOptions) Validate() error {
+	if do.Author == "" {
+		return fmt.Errorf("document author cannot be blank")
+	}
+	return nil
+}
+
+func (do *vexDocOptions) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(
+		&do.DocumentID,
+		"id",
+		"",
+		"ID string for the new VEX document (autogenerated by default)",
+	)
+
+	cmd.PersistentFlags().StringVar(
+		&do.Author,
+		"author",
+		vex.DefaultAuthor,
+		"author to record in the new document",
+	)
+
+	cmd.PersistentFlags().StringVar(
+		&do.AuthorRole,
+		"author-role",
+		vex.DefaultRole,
+		"optional author role to record in the new document",
+	)
+}
+
+type vexStatementOptions struct {
+	Status          string
+	StatusNotes     string
+	Justification   string
+	ImpactStatement string
+	Vulnerability   string
+	ActionStatement string
+	Product         string
+	Subcomponents   []string
+}
+
+// Validate checks that the statement options are coherent
+func (so *vexStatementOptions) Validate() error {
+	if so.Status != string(vex.StatusAffected) && so.ActionStatement == vex.NoActionStatementMsg {
+		so.ActionStatement = ""
+	}
```

Comment:
Ah what that it did was to set the action statement to an empty string when the vex status is different from `affected`, unless you modify it explicitly. 

I think it makes more sense to ignore it completely when status != `affected`. I've changed it to do just that. Also, it will now check to make sure you don't set it when the status is `affected`, as you suggested :+1: